### PR TITLE
Add multi-gpu support to MonaiAlgo

### LINKED ISSUE #5195: Add multi-gpu training support to MonaiAlgo
**Is your feature request related to a problem? Please describe.**
Add multi-gpu training support to MonaiAlgo

**Describe the solution you'd like**
Enable running of MonaiAlgo in a multi-gpu setting.

**Describe alternatives you've considered**
n/a

**Additional context**
n/a

### PR DESCRIPTION
**Pull request by holgerroth**
Signed-off-by: Holger Roth <hroth@nvidia.com>

Fixes #5195.

### Description

Add support for MonaiAlgo to be run with torchrun for multi-gpu training.

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests  --disttests`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

File: monai/networks/utils.py [hunk 1]
```diff
@@ -516,6 +516,8 @@ def copy_model_state(
     unchanged_keys = sorted(set(all_keys).difference(updated_keys))
     logger.info(f"'dst' model updated: {len(updated_keys)} of {len(dst_dict)} variables.")
     if inplace and isinstance(dst, torch.nn.Module):
+        if isinstance(dst, (nn.DataParallel, nn.parallel.DistributedDataParallel)):
```
Comment (Nic-Ma) [inline monai/networks/utils.py:519]: I think line 496 already covered this logic:
```py
dst_dict = OrderedDict(get_state_dict(dst))
```

Thanks.

Comment (wyli) [reply monai/networks/utils.py:519]: This is to modify dst instead of dst_dict, not sure how 496 is useful here...

Comment (Nic-Ma) [reply monai/networks/utils.py:519]: Oh, I misunderstood the logic..

Thanks.

**Review by Nic-Ma — APPROVED** [review]

**Pull request merged**